controlplane/eks/api/v1beta1/awsmanagedcontrolplane_webhook.go

```diff
@@ -337,6 +338,10 @@ func (r *AWSManagedControlPlane) validateDisableVPCCNI() field.ErrorList {
 	if r.Spec.DisableVPCCNI {
 		disableField := field.NewPath("spec", "disableVPCCNI")
 
+		if r.Spec.VpcCni.Disable != r.Spec.DisableVPCCNI {
```

Comment:
I'm struggling to think of the best way to handle this, there's a bit of a dilemna with being able to handle both the new field and the old field.

Currently, `disableVPCCNI` is a bool and it defaults to `false`, this means if the user configures their `AWSManagedControlPlane` like below, it will always error.
```yaml
  # disableVPCCNI: false
  vpcCni:
    disable: true
```

The only solution I can see to this is have the new `disable` field completely override `disableVPCCNI` if it's been set, any thoughts on this?

Reply:
I would make it into a pointer. From the outside, nothing would change. But internally, you can then see if it was just left out or deliberately set to false. In case it was just left out, you can continue with the VpcCni value. WDYT?

Reply:
Sorry for the slow reply @Skarlso I've been away. I also thought about using a pointer, but even if we use a pointer we still have the problem of the default value of `disableVPCCNI` being `false`